src/bin/util/softhsm2-util-botan.cpp

```diff
@@ -68,15 +68,17 @@ int crypto_import_aes_key
 	size_t objIDLen
 )
 {
-	const size_t cMaxAesKeySize = 1024 + 1; // including null-character
+	const size_t cMaxAesKeySize = 1024;
 	char aesKeyValue[cMaxAesKeySize];
+	size_t aesKeyLength = 0;
 	FILE* fp = fopen(filePath, "rb");
```

Comment:
You are right, in [posix](https://pubs.opengroup.org/onlinepubs/9699919799.2016edition/functions/fopen.html) :`The character 'b' shall have no effect, but is allowed for ISO C standard conformance.` 
It seems that even with a `b` on windows the fgets would fail in your case? I'm not a windows user myself.

Reply:
I believe so, but also because `fgets` and `strlen` were used below instead of `fread`.

Reply:
Do we have test cases for the import function? If not, perhaps add one so this code is actually executed.

Reply:
Currently not, because its part of the `src/bin/util` utils not the library itself where the test cases seem to be.